### PR TITLE
Make the Ci tests run on github actions

### PR DESCRIPTION
**Pull request by ryanlerch**
builds on https://github.com/fedora-infra/bodhi/pull/4353

This updates the CI tests to run using github actions. it also removes the old jenkins and zuul configs.

Resolves: #4357 #4248

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  pull_request:
+    branches:
+      - develop
+
+name: Run tests
+
+jobs:
+
+  precommit:
+    name: Run the precommit checks
```
Comment (abompard) [inline .github/workflows/tests.yml]: Maybe call it "Pre-commit checks", to be consistent with the rest?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        run: pip install click conu pytest pytest-cov munch psycopg2 fedora-messaging
+
+      # we only really need to run the precommit checks for one release,
+      # so we just do it for pip
+      - name: Run the precommit checks
+        run: devel/ci/bodhi-ci pre-commit -r pip
+  
+  
+  diff_cover:
+    name: Run the Diff Cover checks
```
Comment (abompard) [inline .github/workflows/tests.yml]: Maybe call it "Diff-cover check", to be consistent with the rest?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ryanlerch) [reply .github/workflows/tests.yml]: updated. also updated the mergify commit to reflect this change

File: .github/workflows/tests.yml (hunk 2 above)
Comment (ryanlerch) [reply .github/workflows/tests.yml]: updated. also updated the mergify commit to reflect this change

**Comment by abompard** [comment]
The unit test fail on bodhi-server for f34, f35 and rawhide but appear as green, and it seems simple to fix, but it reveals what is probably a bug of mine on bodhi-ci, maybe it exits with a zero code even if the tests fail. I'll check.

File: devel/ci/bodhi_ci/unit.py [hunk 3]
```diff
@@ -47,13 +48,13 @@ def __init__(self, *args, **kwargs):
...
-            # Run setup.py develop in each submodule
-            f'for submodule in {modules}; do '
+            # Run setup.py develop in all 3 submodules
+            f'for submodule in {" ".join(MODULES)}; do '
             'cd $submodule; /usr/bin/python3 setup.py develop; cd ..; done; '
             # Run the tests in each submodule
             f'for submodule in {modules}; do '
             'mkdir -p /results/$submodule; '
-            f'cd $submodule; /usr/bin/python3 -m pytest {pytest_flags} || '
+            f'cd $submodule; /usr/bin/python3 -m pytest {pytest_flags}; exit $? || '
```
Comment (abompard) [inline devel/ci/bodhi_ci/unit.py]: Nope, that can't work, it'll exit after the first module every time

**Review by abompard — CHANGES_REQUESTED** [review]

File: devel/ci/bodhi_ci/unit.py (hunk 3 above)
Comment (abompard) [reply devel/ci/bodhi_ci/unit.py]: But it's the right place to fix it! :-) I have fixed it in devel, I'll push it and you can just rebase.

Comment (ryanlerch) [reply devel/ci/bodhi_ci/unit.py]: done, removed my last commit and rebased with your fix

**Review by abompard — APPROVED** [review]

**Pull request merged**